### DISCUSSION TITLE
Prevent Elements  added to canvas on some condition

### DISCUSSION THREAD

**Discussion by saravntbe** [clientIO/joint#2293] [Q&A]
### Introduction

 is it possible to Prevent specific Elements  added to canvas ?

 right now am doing validation on  `graph.on('add', cb)`

```js
const cb = (cell) => {
  if (somecondition) {
    cell.remove();
  }
};
```

if i use `validator.validate('add', cb)` when that callback get executed, during that time Element is already added.

```js
const cb = (err, cmd, next) => {
const el = graph.getElements();  // this includes currently added element as well
if (somecondition) {
    next('Some Error Message');
  }
}
```

 is there a better approach ?

 ---
 my use case:
 - i need to allow only n number of a specific element
 - when some one adds a Element, during that time i need to get all the Elements from canvas & if that specific element exceeds n, then i should not all that element to be added to the canvas.

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment]
There is nothing wrong with using the Validator in this way, but if you want to prevent elements from being added to the diagram and then validated and removed, you can do this instead:

- listen to `element:dragend` stencil event and check the number of elements here and cancel the drag with `stencil.cancelDrag()`
- prevent the user from dragging the stencil element with `validateDrag` stencil callback
- remove the elements from the stencil if they exceeds the total number of instances, effectively prevent the user from dragging them (See [Finite Object Pool](https://www.jointjs.com/demos/finite-object-pool) example).
